killercoda/intro/text.md

```diff
@@ -0,0 +1,26 @@
+# xk6-disruptor interactive demo
+
+```
+          /\      |‾‾| /‾‾/   /‾‾/
+     /\  /  \     |  |/  /   /  /
+    /  \/    \    |     (   /   ‾‾\
+   /          \   |  |\  \ |  (‾)  |
+  / __________ \  |__| \__\ \_____/ .io
+```
+
+Thanks for trying out xk6-disruptor! We're setting up the `xk6-disruptor` tool and a demo application for you. Hold tight while get everything ready!
+
+During this interactive demo, we will:
+
+1. Install the microservice-powered [sock shop](https://github.com/microservices-demo/microservices-demo) demo application.
+2. Make a simple test to understand how the demo application works.
+3. Perform a baseline test against the demo application.
+4. Inject a fault on a backend service and observe how the frontend service behaves.
+
+Step one is being taking care of on the right panel as you read this. After the deployment has been initiated successfully, you can check the deployment progress by running:
+
+```
```

Comment:
I thought about this, but I don't like the idea of locking the user to wait without any indication of progress (which `kubectl wait` arguably does not provide).

The current solution is not perfect either, but an upside of it can be that we give the user an opportunity to get familiar with the killercoda UI (click commands, which do things) while they wait for the deployment to complete.

Reply:
As an additional note, I just checked that some of the pods (e.g. `catalogue`) took 3 good minutes to clear its readiness probe. I don't think it's good UX to have the user wait 3 minutes for a command to complete. Polling status seems a tiny bit more fun, at least.

Reply:
That's part of the reason why I made step 1 ask the user to `curl` the service, so it becomes obvious if something is not working before jumping into k6.

I'll play a bit with the scenario checks to see if it is possible to do something that is not very disruptive.

Reply:
> verify step on the intro section is ignored, it needs to be on a step. This is not a huge problem though.

It seems we have a misunderstanding here. I meant a validation not in the sense of the killercoda scenario definition, but simply as a step we ask the user to execute repeatedly until it returns a successful response.

>That's part of the reason why I made step 1 ask the user to curl the service, so it becomes obvious if something is not working before jumping into k6.

I don't think it is obvious because it can be a "connection reset by peer" or something else, that is why I prefer to explicitly check the pods are ready before invoking the endpoint. 

However, we can leave as it is now and try to find a better way for checking.